src/shared/util.js

```diff
@@ -1269,6 +1364,75 @@ function MessageHandler(sourceName, targetName, comObj) {
             error: reason
           });
         });
+      } else if (data.streamId) {
+        let streamId = data.streamId;
+        let desiredSize = data.desiredSize;
+        let sourceName = this.sourceName;
+        let targetName = data.sourceName;
+        let capability = createPromiseCapability();
+        let streamSink = {
+          enqueue(chunk, size) {
+            var lastDesiredSize = this.desiredSize;
+            this.desiredSize -= size !== undefined ? size : 1;
```

Comment:
Why not change the signature to `enqueue(chunk, size = 1) {` instead, since this line can then be simplified to `this.desiredSize -= size`?